bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/BugsnagReactNativePlugin.kt

```diff
@@ -0,0 +1,104 @@
+package com.bugsnag.android
+
+import java.util.Locale
+
+@Suppress("unused", "UNUSED_PARAMETER")
+class BugsnagReactNativePlugin : Plugin {
+
+    private val configSerializer = ConfigSerializer()
+    private val appSerializer = AppSerializer()
+    private val deviceSerializer = DeviceSerializer()
+    private val breadcrumbSerializer = BreadcrumbSerializer()
+    private val threadSerializer = ThreadSerializer()
+
+    internal lateinit var internalHooks: InternalHooks
+    internal lateinit var client: Client
+
+    lateinit var logger: Logger
+
+    override fun load(client: Client) {
+        client.logger.i("Initialized React Native Plugin")
+    }
+
+    override fun unload() {}
+
+    fun configure(): Map<String, Any?> {
```

Comment:
Can these configure steps to be done in the `load` function? Then expose a `getConfig` method for the map?

Reply:
I don't think we should do this as it complicates the implementation and effectively adds two entrypoints that could interact in unexpected ways. `configure()` will always be the first method called by the bugsnag-js layer so in my mind it makes sense to gather information at the time we need it. This is also part of the design that has been reviewed + approved so I think any architectural change here would need re-review from everyone to make sure it works on all platforms and scenarios.